### ISSUE TITLE
2.3.13 should have been 3.0.0

### ISSUE THREAD

**Issue by gsemet** [vgrem/Office365-REST-Python-Client#525]
Hi

You latest release 2.3.13 breaks some of its structures (`files.file_collection.FileCollection` => `files.collection.FileCollection`). This would have required, if semver was followed, to update to 3.0.0 to signal a breaking change. It is pretty easy to implement, for instance I follow Convetional Commit https://www.conventionalcommits.org/en/v1.0.0/ to track major/minor/breaking changes in my git commits.

We have install that failed this morning because a pip/pipx install checked out your newer version. Thanks anyway for the lib, it makes interfacing with SharePoint a bit less painful.

**Comment by vgrem** [comment]
You're right, due to refactoring (i.e. file renaming) should had probably at least been released under `2.4.0` version, sorry for the inconvenience. `3.0.0` still reserved for release where support for Python 2.7 will be dropped.

**Comment by vgrem** [comment]
Closing - [v3.0.0](https://github.com/vgrem/office365-rest-python-client/releases/tag/v3.0.0) is now live on [[PyPI](https://pypi.org/project/Office365-REST-Python-Client/3.0.0/)](https://pypi.org/project/Office365-REST-Python-Client/3.0.0/). The semver concern raised here is resolved - breaking changes (legacy auth removal, Python 2 drop, module restructuring) are now properly signalled with a major version bump.

Going forward, [[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)](https://www.conventionalcommits.org/en/v1.0.0/) is a good suggestion - pairing it with the automated release pipeline now in place would make semver enforcement automatic. PRs with `feat!:` or `BREAKING CHANGE:` in commits would trigger a major version bump via tooling like `semantic-release` or `release-please`.

```bash
pip install --upgrade office365-rest-python-client   # installs 3.0.0
```

Pin to a major version to avoid future surprises:
```
office365-rest-python-client>=3.0.0,<4.0.0
```
